streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/system/service/impl/VariableServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+public class VariableServiceImpl extends ServiceImpl<VariableMapper, Variable> implements VariableService {
+
+    @Autowired
+    private ApplicationService applicationService;
+
+    @Autowired
+    private CommonService commonService;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public void createVariable(Variable variable) throws Exception {
+        if (isExists(variable)) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        variable.setCreator(commonService.getCurrentUser().getUserId());
+        this.save(variable);
+    }
+
+    @Override
+    public IPage<Variable> findVariables(Variable variable, RestRequest request) {
+        Page<Variable> page = new Page<>();
+        page.setCurrent(request.getPageNum());
+        page.setSize(request.getPageSize());
+        return this.baseMapper.findVariable(page, variable);
+    }
+
+    @Override
+    public void updateVariable(Variable variable) throws Exception {
+        Variable findVariable = this.findByVariableCode(variable.getTeamId(), variable.getVariableCode());
+        if (findVariable == null) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        if (findVariable.getId().longValue() != variable.getId().longValue()) {
+            throw new ApiAlertException("Sorry, the variable id is inconsistent.");
+        }
+        updateById(variable);
+    }
+
+    @Override
+    public void deleteVariable(Variable variable) throws Exception {
+        Variable findVariable = this.findByVariableCode(variable.getTeamId(), variable.getVariableCode());
+        if (findVariable == null) {
+            throw new ApiAlertException("Sorry, the variable code already exists.");
+        }
+        if (findVariable.getId().longValue() != variable.getId().longValue()) {
+            throw new ApiAlertException("Sorry, the variable id is inconsistent.");
+        }
+        List<Application> dependApplications = this.findDependByCode(variable);
+        if (!(dependApplications == null || dependApplications.isEmpty())) {
+            throw new ApiAlertException(String.format("Sorry, this variable is being used by [%s] applications.", dependApplications.size()));
+        }
+        this.removeById(findVariable.getId());
+    }
+
+    @Override
+    public Variable findByVariableCode(Long teamId, String variableCode) {
+        return baseMapper.selectOne(new LambdaQueryWrapper<Variable>()
+            .eq(Variable::getVariableCode, variableCode)
+            .eq(Variable::getTeamId, teamId));
+    }
+
+    @Override
+    public List<Variable> findByTeamId(Long teamId) {
+        return baseMapper.selectByTeamId(teamId);
+    }
+
+    @Override
+    public boolean isExists(Variable variable) {
+        return this.findByVariableCode(variable.getTeamId(), variable.getVariableCode()) != null;
+    }
```

Comment:
Could you delete this method? The method name isn't clear.

If you don't see the detailed code and just see the method name, do you know how this method checks if the variable exists? I see the method name, I thought it check by `variable.getId()`

Ambiguous method names introduce potential risks and bugs. It can easily lead to bugs if called by other developers.

Reply:
I deleted it.